src/app/cli/src/commands/system_diagnose_command.rs

```diff
@@ -0,0 +1,197 @@
+            .container_runtime
+            .custom_cmd("--version".to_string())
+            .output()
+            .await
+        {
+            Ok(_) => style("ok").green(),
+            Err(_) => style("failed").red(),
+        };
+
+        write!(f, "{is_container_installed}")?;
+        Ok(())
+    }
+
+    async fn run_container_pull_check(&self, f: &mut Stdout) -> Result<(), CLIError> {
+        write!(f, "\ncontainer can pull images... ")?;
+
+        let is_container_image_pulled =
+            match self.container_runtime.pull_image(DUMMY_IMAGE, None).await {
+                Ok(_) => style("ok").green(),
+                Err(_) => style("failed").red(),
+            };
+
+        write!(f, "{is_container_image_pulled}")?;
+        Ok(())
+    }
+
+    async fn run_container_rootless_check(&self, f: &mut Stdout) -> Result<(), CLIError> {
+        write!(f, "\ncontainer rootless check... ")?;
+
+        let is_container_rootless = match self
+            .container_runtime
+            .run_attached(DUMMY_IMAGE)
+            .init(true)
+            .spawn()
+            .unwrap()
+            .wait()
+            .await
+        {
+            Ok(_) => style("ok").green(),
+            Err(_) => style("failed").red(),
+        };
+
+        write!(f, "{is_container_rootless}")?;
+        Ok(())
+    }
+
+    async fn run_container_volume_mount_check(&self, f: &mut Stdout) -> Result<(), CLIError> {
+        write!(f, "\ncontainer volume mounts work... ")?;
+        let temp_dir = tempfile::tempdir()?;
+        let cwd = Path::new(".").canonicalize()?;
+
+        let file_path = temp_dir.path().join("tmp.txt");
+        let _ = File::create(file_path)?;
+
+        let is_container_rootless = match self
+            .container_runtime
+            .run_attached(DUMMY_IMAGE)
+            .volume((cwd, temp_dir.into_path()))
+            .init(true)
+            .spawn()
+            .unwrap()
+            .wait()
+            .await
+        {
+            Ok(_) => style("ok").green(),
+            Err(_) => style("failed").red(),
+        };
+
+        write!(f, "{is_container_rootless}")?;
+        Ok(())
+    }
+
+    async fn run_workspace_consistency_check(
+        &self,
+        dataset_repo: Arc<dyn DatasetRepository>,
+        verification_svc: Arc<dyn VerificationService>,
+        f: &mut Stdout,
+    ) -> Result<(), CLIError> {
+        write!(f, "\nworkspace consistent... ")?;
+        let datasets: Vec<_> = dataset_repo.get_all_datasets().try_collect().await?;
+
+        let verify_options = VerificationOptions {
+            check_integrity: true,
+            replay_transformations: false,
+        };
+        for dataset in datasets {
+            if let Err(_) = verification_svc
+                .verify(
+                    &dataset.as_local_ref(),
+                    (None, None),
+                    verify_options.clone(),
+                    None,
+                )
+                .await
```

Comment:
For a large enough workspace, this is not printing any progress for a long time.
As a minimum, we must indicate the dataset we started to check.

But also, the verification should show some progress indication. See how `kamu verify` command displays it is busy - implement something similar here too, otherwise I would kill the diagnosis faster than it finishes.